src/main/java/cz/centrum/haffner/SimpleTrainingDavid/AppServices/OneDayFileJsonParser.java

```diff
@@ -0,0 +1,122 @@
+package cz.centrum.haffner.SimpleTrainingDavid.AppServices;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import cz.centrum.haffner.SimpleTrainingDavid.DataTemplates.MetricsInfoDataSet;
+import org.springframework.stereotype.Component;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class OneDayFileJsonParser {
+    // TODO: net approach doesn´t work
+    // private static final String FILE_PATH = "https://github.com/DavidHaffner/SimpleTrainingDavid/tree/dev/logs/MCP_20180131.json";
+    private static final String FILE_PATH_PREFIX = "C://work/SimpleTrainingDavid/src/main/resources/MCPData/";
+
+    private int callsCounter = 0;
+    private int okCallsCounter = 0;
+    private int koCallsCounter = 0;
+
+
+    public OneDayFileJsonParser() {
+    }
+
+
+    public MetricsInfoDataSet parseOneDayFile (long requestedDate, MetricsInfoDataSet metricsInfoDataSet) {
```

Comment:
Když parsuješ nějaký file - nedávalo by smysl do této metody už posílat třeba datovou strukturu "File" z MetricsInfoService a nad tou si poté vytvořit FileReader atd..?
Představ si, že tahle metoda parsuje stejně strukturované soubory, které ale můžou přícházet z více různých míst a budou mít jiný FILE_PATH_PREFIX a třeba i nějaký suffix a vůbec, úplně jinou cestu k nim. (každá servisa, která by tuto metodu volala, by načítala soubory z jiných umístění).
Proto pokud metoda slouží pro parsování souboru - předávej rovnou odkaz na soubor a ten vytvoř v byznys logice servisy.

Reply:
done